flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/io/StreamTwoInputProcessor.java

```diff
@@ -54,92 +50,98 @@
-		String taskName,
-		OperatorChain<?, ?> operatorChain,
-		Counter numRecordsIn) throws IOException {
-
-		this.streamOperator = checkNotNull(streamOperator);
-		this.inputSelectionHandler = checkNotNull(inputSelectionHandler);
+			CheckpointedInputGate[] checkpointedInputGates,
+			TypeSerializer<IN1> inputSerializer1,
+			TypeSerializer<IN2> inputSerializer2,
+			Object lock,
+			IOManager ioManager,
+			StreamStatusMaintainer streamStatusMaintainer,
+			TwoInputStreamOperator<IN1, IN2, ?> streamOperator,
+			TwoInputSelectionHandler inputSelectionHandler,
+			WatermarkGauge input1WatermarkGauge,
+			WatermarkGauge input2WatermarkGauge,
+			OperatorChain<?, ?> operatorChain,
+			Counter numRecordsIn) {
 
 		this.lock = checkNotNull(lock);
+		this.inputSelectionHandler = checkNotNull(inputSelectionHandler);
 
-		InputGate unionedInputGate1 = InputGateUtil.createInputGate(inputGates1.toArray(new InputGate[0]));
-		InputGate unionedInputGate2 = InputGateUtil.createInputGate(inputGates2.toArray(new InputGate[0]));
-
-		// create a Input instance for each input
-		CheckpointedInputGate[] checkpointedInputGates = InputProcessorUtil.createCheckpointedInputGatePair(
-			streamTask,
-			checkpointingMode,
-			ioManager,
-			unionedInputGate1,
-			unionedInputGate2,
-			taskManagerConfig,
-			taskName);
-		checkState(checkpointedInputGates.length == 2);
-		this.input1 = new StreamTaskNetworkInput(checkpointedInputGates[0], inputSerializer1, ioManager, 0);
-		this.input2 = new StreamTaskNetworkInput(checkpointedInputGates[1], inputSerializer2, ioManager, 1);
-
-		this.statusWatermarkValve1 = new StatusWatermarkValve(
-			unionedInputGate1.getNumberOfInputChannels(),
-			new ForwardingValveOutputHandler(streamOperator, lock, streamStatusMaintainer, input1WatermarkGauge, 0));
-		this.statusWatermarkValve2 = new StatusWatermarkValve(
-			unionedInputGate2.getNumberOfInputChannels(),
-			new ForwardingValveOutputHandler(streamOperator, lock, streamStatusMaintainer, input2WatermarkGauge, 1));
+		this.output1 = new StreamTaskNetworkOutput<>(
+			streamOperator,
+			record -> processRecord1(record, streamOperator, numRecordsIn),
+			lock,
+			streamStatusMaintainer,
+			input1WatermarkGauge,
+			0);
+
+		this.output2 = new StreamTaskNetworkOutput<>(
+			streamOperator,
+			record -> processRecord2(record, streamOperator, numRecordsIn),
+			lock,
+			streamStatusMaintainer,
+			input2WatermarkGauge,
+			1);
+
+		StatusWatermarkValve statusWatermarkValve1 = new StatusWatermarkValve(
+			checkpointedInputGates[0].getNumberOfInputChannels(), output1);
+		StatusWatermarkValve statusWatermarkValve2 = new StatusWatermarkValve(
+			checkpointedInputGates[1].getNumberOfInputChannels(), output2);
+
+		this.input1 = new StreamTaskNetworkInput<>(
+			checkpointedInputGates[0], inputSerializer1, ioManager, statusWatermarkValve1, 0);
+		this.input2 = new StreamTaskNetworkInput<>(
+			checkpointedInputGates[1], inputSerializer2, ioManager, statusWatermarkValve2, 1);
```

Comment:
For code formatting: this and above code, related for arguments on new lines (see: https://flink.apache.org/contributing/code-style-and-quality-formatting.html#breaking-the-lines-of-too-long-statements).

Reply:
Thanks. In my previous memories, it was both ok for arguments either in the same line or in separate line. Now the new formatting seems only accept the separate argument line if breaking. I should catch up with it and fix based on that. :)